www/docs/client/links/httpSubscriptionLink.md

```diff
@@ -251,6 +251,88 @@ const trpc = createTRPCClient<AppRouter>({
 });
 ```
 
+## Timeout Configuration {#timeout}
+
+The `httpSubscriptionLink` supports configuring a timeout for inactivity through the `reconnectAfterInactivityMs` option. If no messages (including ping messages) are received within the specified timeout period, the connection will be marked as "connecting" and automatically attempt to reconnect.
+
+```ts
+import {
+  createTRPCClient,
+  httpBatchLink,
+  splitLink,
+  unstable_httpSubscriptionLink,
+} from '@trpc/client';
+
+const trpc = createTRPCClient<AppRouter>({
+  links: [
+    splitLink({
+      condition: (op) => op.type === 'subscription',
+      true: unstable_httpSubscriptionLink({
+        url: '/api/trpc',
+        // Reconnect if no messages received for 30 seconds
+        reconnectAfterInactivityMs: 30_000,
+      }),
+      false: httpBatchLink({
+        url: '/api/trpc',
+      }),
+    }),
+  ],
+});
+```
+
+When a timeout occurs:
+
+1. The connection state will change to "connecting"
+2. The client will automatically attempt to reconnect
+3. Any tracked event IDs will be preserved and sent with the new connection to resume from the last received message
+
+This is useful for:
+
+- Detecting stale connections that may have silently failed
+- Automatically recovering from network interruptions
+- Ensuring your subscriptions stay active even during periods of inactivity
+
+:::tip
+The timeout includes ping messages sent by the server. If you're experiencing timeouts during normal operation, you may want to:
+
+1. Increase the `reconnectAfterInactivityMs` value
+2. Adjust your server's ping interval to be more frequent
+
+:::
+
+## Server Ping Configuration {#server-ping}
+
+The server can be configured to send periodic ping messages to keep the connection alive and prevent timeout disconnections. This is particularly useful when combined with the `reconnectAfterInactivityMs` option on the client.
+
+```ts title="server/trpc.ts"
+import { initTRPC } from '@trpc/server';
+
+export const t = initTRPC.create({
+  sse: {
+    // Maximum duration of a single SSE connection in milliseconds
+    // maxDurationMs: 60_00,
+    ping: {
+      // Enable periodic ping messages to keep connection alive
+      enabled: true,
+      // Send ping message every 200ms
+      intervalMs: 200,
+    },
+  },
+});
+```
+
+When configuring both client timeout and server ping, ensure that:
```

Comment:
could this be a validation step at runtime?

Reply:
tricky tbh